### ISSUE TITLE
TOC incorrect link to header that contains vertical bar character

### ISSUE THREAD

**Issue by cosinekitty** [thlorenz/doctoc#71]
Very useful tool!  I did find a minor issue.  In a Github wiki page I have a header that looks like this:

    #### `history [pgn | alg]`

When I open the wiki page in my browser and View Source, I see this where the header is in the html code:

````
<h4>
<a id="user-content-history-pgn--alg" class="anchor" href="#history-pgn--alg" aria-hidden="true"><span class="octicon octicon-link"></span></a><code>history [pgn | alg]</code>
</h4>
````

DocToc generates the following TOC entry:

    - [`history [pgn | alg]`](#history-pgn-%7C-alg)

Note the TOC link's anchor tag has an extraneous `%7C` in it.

I am manually fixing this for now, but I thought you would want to know.

Update: A similar problem happens when the header title text contains a `#` character: GitHub's Markdown processor creates an anchor without the `#`, but that character is still in DocToc's link to the anchor.

**Comment by jez** [comment]
It looks like this is actually an issue with [thlorenz/anchor-markdown-header](https://github.com/thlorenz/anchor-markdown-header).

It also looks like our code treats GitHub and BitBucket somewhat similarly in the code. Looking at how BitBucket handles the case you outlined, they actually output `#history-pgn-alg`, presumably collapsing the consecutive hyphens.

**Comment by jez** [comment]
@thlorenz I've sent a patch to the upstream repo that fixes this issue.

**Comment by cosinekitty** [comment]
Thanks you @jez .
I looked at the code change.  It looks like there will still be a problem with the `#` character in a title though.  Example:

````
## Demo #1: using the `-s` option
````

The correct link should be the following:

````
- [Demo #1: using the `-s` option](#demo-1-using-the--s-option)
````

The link actually generated by DocToc was:

````
- [Demo #1: using the `-s` option](#demo-#1-using-the--s-option)
````

- Don

**Comment by jez** [comment]
This should be addressed in jez/anchor-markdown-header@3b7210a in the PR thlorenz/anchor-markdown-header#24.

**Comment by jez** [comment]
@thlorenz I'm not sure what the best way to bump the versions of the deps is, but it should be the only thing left to do.

**Comment by thlorenz** [comment]
It's a patch update so it'll be pulled automagically due to [the `~`](https://github.com/thlorenz/doctoc/blob/master/package.json#L19) -- would also work for minor updated.

People just have to reinstall doctoc to get it:

```
npm rm -g doctoc && npm i -g doctoc
```

**Comment by jez** [comment]
Oh ok, cool.

It might be useful to peg anchor-markdown-header at 0.5.4 to force an
update (for example, npm outdated won't list doctoc as being outdated).

**Comment by thlorenz** [comment]
I don't peg versions usually, but if you wanna just change the version to `^0.5.4` that'd be fine with me.

**Comment by jez** [comment]
Closed by #72.
